packages/outstatic/src/utils/yup.ts

```diff
@@ -27,4 +28,26 @@ export const editDocumentSchema: yup.SchemaOf<Document> = yup.object().shape({
     .required(),
   description: yup.string(),
   coverImage: yup.string()
-})
+}
+
+export const editDocumentSchema: yup.SchemaOf<SchemaShape> = yup
+  .object()
+  .shape(documentShape)
+
+export const convertSchemaToYup = (customFields: {
+  properties: CustomFields
+}) => {
+  const shape: SchemaShape = {}
+
+  Object.entries(customFields.properties).map(([name, fields]) => {
```

Comment:
Do you think the need for custom coercion will come up regularly? This section could get pretty wordy with more custom types.

Reply:
Fixed this by storing data type and field type in json.